examples/with-thirdparty/app/src/main/java/com/supertokens/supertokensexample/LoginActivity.kt

```diff
@@ -0,0 +1,195 @@
+package com.supertokens.supertokensexample
+
+import android.content.Intent
+import android.content.Intent.FLAG_ACTIVITY_CLEAR_TASK
+import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.AppCompatButton
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.SignInButton
+import com.supertokens.supertokensexample.services.network.APIManager
+import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationRequest
+import net.openid.appauth.AuthorizationResponse
+import net.openid.appauth.AuthorizationService
+import net.openid.appauth.AuthorizationServiceConfiguration
+import net.openid.appauth.ResponseTypeValues
+import okhttp3.MediaType
+import okhttp3.RequestBody
+import okhttp3.ResponseBody
+import org.json.JSONObject
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var googleResultLauncher: ActivityResultLauncher<Intent>
+    private lateinit var githubResultLauncher: ActivityResultLauncher<Intent>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+
+        val googleButton = findViewById<SignInButton>(R.id.btGoogle)
+        val githubButton = findViewById<AppCompatButton>(R.id.btGithub)
+
+        googleButton.setOnClickListener {
+            signInWithGoogle()
+        }
+
+        githubButton.setOnClickListener {
+            signInWithGitHub()
+        }
+
+        googleResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+            onGoogleResultReceived(it)
+        }
+
+        githubResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+            onGithubResultReceived(it)
+        }
+    }
+
+    private fun signInWithGoogle() {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken("GOOGLE_WEB_CLIENT_ID")
+            .requestEmail()
+            .build()
+
+        val googleClient = GoogleSignIn.getClient(this, gso)
+        val signInIntent = googleClient.signInIntent
+
+        googleResultLauncher.launch(signInIntent)
+    }
+
+    private fun signInWithGitHub() {
+        val serviceConfig = AuthorizationServiceConfiguration(Uri.parse("https://github.com/login/oauth/authorize"), Uri.parse("https://github.com/login/oauth/access_token"))
```

Comment:
you should call the backedn's getAuthorisationUrl API to get the URl to redirect to right?

Reply:
The library we use requires a native screen which takes over the web redirection. Ive explained this in the README

```
Use `com.supertokens.supertokensexample://oauthredirect` when configuring the Authorization callback URL.
```

In the line above we just need the authorization endpoint and the token endpoint.